File: runtime/cpp/titan-ast-runtime-lib/StringUtils.cpp
//
// for ascll
// Created by tian wei jun on 2022/11/24 0024.
//
//

#include "StringUtils.h"

bool StringUtils::isEmpty(const std::string *str) {
  return !str || str->length() <= 0;
}

bool StringUtils::isNotEmpty(const std::string *str) { return !isEmpty(str); }

bool StringUtils::isBlank(const std::string *str) {
  if (!str || str->length() <= 0) {
    return true;
  }
  bool isBlank = true;
  const char *chars = str->data();
  const char *charIt = chars;
  for (int i = 0; i < str->length(); i++) {
    char tmpChar = *charIt;
    switch (tmpChar) {// {32, 9, 13}
      case 32:
      case 9:
      case 13:
        break;
      default:
        isBlank = false;
    }
    if (!isBlank) {
      break;
    }
    ++charIt;
  }
  return isBlank;
}

bool StringUtils::isNotBlank(const std::string *str) { return !isBlank(str); }

File: runtime/java/titan-ast-runtime/src/titan/ast/runtime/StringUtils.java
package titan.ast.runtime;

/**
 * StringUtils.
 *
 * @author tian wei jun
 */
public class StringUtils {

  public static boolean isEmpty(String str) {
    return str == null || str.length() == 0;
  }

  public static boolean isNotEmpty(String str) {
    return !StringUtils.isEmpty(str);
  }

  /**
   * 判断字符串是不是空白符.
   *
   * @param str 字符串
   * @return 字符串是不是空白符,是的话返回true否则false
   */
  public static boolean isBlank(String str) {
    int strLen;
    if (str == null || (strLen = str.length()) == 0) {
      return true;
    }
    for (int indexOfChar = 0; indexOfChar < strLen; indexOfChar++) {
      char ch = str.charAt(indexOfChar);
      if (!Character.isWhitespace(ch)) {
        return false;
      }
    }
    return true;
  }

  public static boolean isNotBlank(String str) {
    return !StringUtils.isBlank(str);
  }
}
